src/backend/core/api/viewsets.py

```diff
@@ -160,8 +160,8 @@ def get_queryset(self):
         if document_id := self.request.GET.get("document_id", ""):
             queryset = queryset.exclude(documentaccess__document_id=document_id)
 
-        if not (query := self.request.GET.get("q", "")):
-            return queryset
+        if not (query := self.request.GET.get("q", "")) or len(query) < 5:
```

Comment:
`3` is I think "less" aggressive. 5 probably match for european or "occidental" names but we all have example of name length of only 1 character (like is asia).